packages/vue-components/src/Searchbar.vue

```diff
@@ -121,7 +121,7 @@ export default {
           headingKeywords,
         } = entry;
         const keywords = entry.keywords || '';
-        const displayTitle = title || src;
+        const displayTitle = title || src.substring(0, src.lastIndexOf('.'));
```

Comment:
Although not a major issue considering how the normal situation would be like, this would give unexpected results when the file has no extensions (like just `index`) or "double" extensions (things like `index.tar.gz`).

The latter seems a bit unlikely to be passed on to the search so I guess it's okay to not cover that (let me know if you feel otherwise though!), but it's good fault-tolerance if we can handle the former case.

Reply:
Okay, I have delved in a little bit on how we include pages and the search, and looks like the latter case (and probably the former too) can be reasonably ignored 😅

Search data comes from `siteData.json`, which is written by `Site`, that includes data from `Site.pages`, which in turn is collected from `site.json` properties (`pages`, `globs`, and the likes). I guess we can reasonably believe that search results come from page files and not any arbitrary file. As our page files are of the normal extension form (`.md` and `.html`), then we can assume that the latter case would not happen.

By extension (heh), the former case shouldn't really happen as well. But I still think it's good to be safe rather than encountering a case where the result are empty strings one day. 😄

Reply:
I investigated this suggestion more as well, and found the following code in NodeProcessor.

`      const fileExt = path.extname(file);
      if (fsUtil.isMarkdownFileExt(fileExt)) {
        const renderedContent = this.markdownProcessor.renderMd(content);
        // Wrap with <root> as $.remove() does not work on top level nodes
        parser.parseComplete(`<root>${renderedContent}</root>`);
      } else if (fileExt === '.html') {
        parser.parseComplete(`<root>${content}</root>`);
      } else {
        const error = new Error(`Unsupported File Extension: '${fileExt}'`);
        reject(error);
      }`

In other words, MarkBind only supports.html or .md . If you deliberately include a 'page' which doesn't have a supported file format ending, the `Unsupported File Extension:` error is thrown and the website cannot build at all.

So I think both the former and the latter case are invalid inputs that will safely never happen under the current implementation 😅

Reply:
Thanks for the deeper investigation @kaixin-hc, we can safely use this implementation then 😄